e2e.go

```diff
@@ -37,26 +45,95 @@ func runE2E(workDir string) error {
 		log.Infof("Using secrets from: %s", secrets)
 	}
 
-	tempDir, err := ioutil.TempDir("", "")
+	workflows, err := readE2EWorkflows(e2eBitriseYMLPath)
 	if err != nil {
-		return fmt.Errorf("failed to create temporary directory: %v", err)
+		return err
+	}
+
+	for _, workflow := range workflows {
+		err = runE2EWorkflow(commandFactory, workDir, e2eBitriseYMLPath, secrets, workflow)
+		if parentURL != "" {
+			if err := sendAnalytics(segmentKey, workflow, err == nil, parentURL); err != nil {
+				return err
+			}
+		}
+		if err != nil {
+			return fmt.Errorf("failed to run workflow %s: %w", workflow, err)
+		}
 	}
 
-	targetConfig := filepath.Join(tempDir, "bitrise_e2e.yml")
+	return nil
+}
 
-	if err := generateE2EWorkflow(e2eBitriseYMLPath, targetConfig); err != nil {
-		return err
+func sendAnalytics(segmentKey string, workflow string, success bool, parentURL string) (err error) {
+	client := analytics.New(segmentKey)
```

Comment:
Is it okay to create a new analytics client every time a workflow is finished? If it doesn't have too much overhead or any other weird behavior then I'm okay with it.

Reply:
good catch, updating